src/components/Home/WelcomeHeader.js

```diff
@@ -37,6 +48,94 @@ const WelcomeHeader = () => {
       },
     },
   };
+  
+  useEffect(() => {
+    if (localPath !== ''){
+      const installSampleProtocol = async() => {
+        return openNetcanvas(localPath)
+      }
+      installSampleProtocol()
+    }
+  }, [localPath])
+
+  const getURL = uri =>
+  new Promise((resolve, reject) => {
+    try {
+      resolve(new URL(uri));
+    } catch (error) {
+      reject(error);
+    }
+  });
+
+const urlError = friendlyErrorMessage("The location you gave us doesn't seem to be a valid URL. Check the location, and try again.");
+const networkError = friendlyErrorMessage("We weren't able to fetch your protocol. Your device may not have an active network connection, or you may have mistyped the URL. Ensure you are connected to a network, double check your URL, and try again.");
+const fileError = friendlyErrorMessage('The protocol could not be saved to your device. You might not have enough storage available. ');
+
+/**
+ * Download a protocol from a remote server.
+ *
+ * If the URL points to an instance of a Network Canvas Server, then the caller must ensure
+ * that the SSL certificate has been trusted. See {@link ApiClient#addTrustedCert}.
+ *
+ * @param {string} uri
+ * @return {string} output filepath
+ */
+const downloadProtocolFromURI = inEnvironment((environment)=> {
+
+  if (environment === environments.ELECTRON) {
+    const request = require('request-promise-native');
```

Comment:
This is an npm dependency that you did not add to the `package.json` of the project. The only reason this code works is because other dependencies import this module. This is very bad for two reasons:

1. You do not determine the specific version of the dependency that is being imported. This means that you do not know the specific API for using the module.
2. Upgrading this dependency is out of your control. If a transient dependency is upgraded, this module version will likely change. It may even be removed entirely, which will cause the software to crash or not build.

All dependencies must be specified in `package.json`!

However, there is a more significant issue here. Remember that I mentioned that there is a great deal of legacy code in the project? Here, the module you have used is marked as no longer supported by its author: https://github.com/request/request-promise-native#deprecated

This means that we should not introduce any additional uses of this module anywhere in our code, and we should refactor any code that is using it when we change it. 

You should think more critically about what a dependency is doing, and determine for yourself if it is needed. In this case, `request-promise-native` is an es6 promise wrapper around `request`. Its purpose is to fetch the requested resource using HTTP GET. This is very basic functionality, and we should think about if we can replace this module with a better method, or a more modern dependency.

Because we are operating in a Chromium/Node environment (Electron), we can choose from several potential replacements. If we want to use node (which would allow us to move this code to the backend in the future), we could use `node-fetch`, `axios`, `got` or similar. However, since this code is running in the renderer thread of Electron, we have access to the chromium browser APIs, including `fetch` (https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch). This is what I suggest you use here.